x-pack/plugins/lens/public/plugin.ts

```diff
@@ -243,7 +243,7 @@ export class LensPlugin {
 
     return {
       EmbeddableComponent: getEmbeddableComponent(startDependencies.embeddable),
-      navigateToPrefilledEditor: (input: LensEmbeddableInput) => {
+      navigateToPrefilledEditor: (input: LensEmbeddableInput, openInNewTab?: boolean) => {
         if (input.timeRange) {
           startDependencies.data.query.timefilter.timefilter.setTime(input.timeRange);
```

Comment:
This doesn't work for a new tab - if you test the example, it will fall back to the default time range.

One way to work around this is putting this piece of information into the path of the URL (this is how it survives page reloads as well).

We have to add a string like this: `?_g=(time:(from:now-7d,to:now))`.

Another option is to use the time range from the input within the Lens app (this is not done right now). Here: https://github.com/elastic/kibana/blob/7984745a9d9a438513cd3f5505149d0d62c529e2/x-pack/plugins/lens/public/app_plugin/app.tsx#L316

`initialInput.timeRange` is set - we could check for it and call `timefilter.setTime` right there.

Not sure right now which option is better - the second one seems cleaner, but we would have to check whether it has unwanted side effects.

Reply:
@flash1293 i moved it to the url, i find the simpler approach.

Reply:
I'm not happy with how it's done in the PR - it can fall out of sync with the rest of Kibana quite easily without anyone noticing:
* `_g` parameter name is hardcoded
* rison serialization is done inline, there might be edge cases that won't work (have you tried absolute dates?)
* No unit tests

Reply:
@shahzad31 for visibility